### ISSUE TITLE
VectorQuantizer Loss

### ISSUE THREAD

**Issue by bochennn** [huggingface/diffusers#8226]
### Describe the bug

I'm using the VQModel in diffusers for my generation project and I got a little confusion on calculating the vector quantizer loss:

https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/autoencoders/vae.py#L731

The first part "torch.mean((z_q.detach() - z) ** 2)" should be the commitment loss, and the second part "torch.mean((z_q - z.detach()) ** 2)" is embedding loss.
But, according to the original paper: https://arxiv.org/abs/1711.00937, the vq loss described as equation(3). The beta weight is only mutiplyed to the commitment loss

![image](https://github.com/huggingface/diffusers/assets/17314925/13a40eb9-fa15-45e5-bd63-2e6cf29bdfd0)

I'm not sure if this was intentional, because I'm aware that the code has been fixed once after adding the "legacy" param. But I'm pretty sure the fixed line is still incorrect.

Here is some of the other implementation I found computing the vq loss which I considered correct:
https://github.com/clementchadebec/benchmark_VAE/blob/main/src/pythae/models/vq_vae/vq_vae_utils.py#L64
https://github.com/myc634/UltraLiDAR_nusc_waymo/blob/master/plugin/models/necks/vq_layer.py#L131

Feel free to talk more about this!

### Reproduction

loss = torch.mean(self.beta * (z_q.detach() - z) ** 2) + torch.mean((z_q - z.detach()) ** 2)

### System Info

- `diffusers` version: 0.27.2
- Platform: Linux-5.15.0-105-generic-x86_64-with-glibc2.29
- Python version: 3.8.10
- PyTorch version (GPU?): 2.0.0+cu118 (True)
- Huggingface_hub version: 0.22.2
- Transformers version: 4.39.3
- Accelerate version: 0.29.2
- xFormers version: not installed
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### Who can help?

@sayakpaul @DN6 @yiyixuxu

**Comment by bochennn** [comment]
Sorry I misread this line. I'll retrieve my issue
